htslib/hts.h

```diff
@@ -538,6 +538,15 @@ HTSLIB_EXPORT
 int hts_set_opt(htsFile *fp, enum hts_fmt_option opt, ...);
 
 HTSLIB_EXPORT
+/*!
+    @abstract        Reads a line of text
+    @param fp        htsFile pointer
+    @param delimiter Line separator char, or KS_SEP_* values (see kseq.h)
```

Comment:
Since 71c03b88edf892a93eae7bb7932a6da84858480a `hts_getline` only works on actual lines, so `delimiter` must be literally `'\n'` or `KS_SEP_LINE` — see the TODO in that commit message.

(`HTSLIB_EXPORT` goes after the comment.)

Reply:
I struggled writing the documentation for this as the operation of it is unclear and, obviously, there's no documentation!  Even the return values were pretty vague and confusing.

Please make a suggestion for revised text as I expect you know what it's meant to be doing far more than I do.

Reply:
It was almost four years ago :smile: but here you go:

```
  @abstract         Read a line (and its \n or \r\n terminator) from a file
  @param fp         The file handle
  @param delimiter  Unused, but must be '\n' (or KS_SEP_LINE)
  @param str        The line (not including the terminator) is written here
  @return           Length of the string read; -1 on end-of-file; <= -2 on error
```

Reply:
Well it was worth asking - I didn't know it's actually unused.  Thanks will update soon.  Working on some write side checks atm.